dhcpv4/nclient4/client.go

```diff
@@ -63,35 +74,61 @@ type pendingCh struct {
 	ch chan<- *dhcpv4.DHCPv4
 }
 
-type logger interface {
-	Printf(format string, v ...interface{})
+// Logger is a handler which will be used to output logging messages
+type Logger interface {
+	// PrintMessage print _all_ DHCP messages
 	PrintMessage(prefix string, message *dhcpv4.DHCPv4)
+
+	// Printf is use to print the rest debugging information
+	Printf(format string, v ...interface{})
 }
 
-type emptyLogger struct{}
+// EmptyLogger prints nothing
+type EmptyLogger struct{}
```

Comment:
Do we need to make this public?

Reply:
I don't see any reason to hide it. Somebody can do now something like:
```
var logger nclient4.Logger
switch loggerType {
case someValue0:
    logger = &nclient4.EmptyLogger{}
case someValue1:
    logger = &nclient4.DebugLogger{Printfer: log.New(...)}
case someValue2:
    logger = &myLogger{}
}
opts = append(opts, nclient4.WithLogger(logger))
```

But I can hide it if you want :)

Reply:
No, it is fine, if you think it may be useful lets leave it.